### PR TITLE
Allow denoise_bilateral to work on images with negative values

### PR DESCRIPTION
**Pull request by grlee77**
## Description

This PR revives #1679, fixing merge conflicts and addressing style comments made there.

I also made a small modification so that the minimum is only subtracted when it is negative to avoid any overhead for positive-valued images.

## Checklist

- [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- Gallery example in `./doc/examples` (new features only)
- Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- Unit tests
- Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)

## For reviewers

- Check that the PR title is short, concise, and will make sense 1 year
  later.
- Check that new functions are imported in corresponding `__init__.py`.
- Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.

### CODE REVIEW

File: skimage/restoration/_denoise.py [hunk 1]
```diff
@@ -222,8 +222,6 @@ def denoise_bilateral(image, win_size=None, sigma_color=None, sigma_spatial=1,
     # if image.max() is 0, then dist_scale can have an unverified value
     # and color_lut[<int>(dist * dist_scale)] may cause a segmentation fault
     # so we verify we have a positive image and that the max is not 0.0.
-    if min_value < 0.0:
-        raise ValueError("Image must contain only positive values")
 
     if max_value == 0.0:
         raise ValueError("The maximum value found in the image was 0.")
```
Comment (rfezzani) [inline skimage/restoration/_denoise.py:227]:
```suggestion
```
This is no more relevant if we want to support negative values :wink:

Comment (grlee77) [reply skimage/restoration/_denoise.py]: Thanks

**Review by rfezzani — APPROVED** [review]
Thank you @grlee77

**Comment by alexdesiqueira** [comment]
Thank you @grlee77!

**Pull request merged**